packages/expo-notifications/ios/EXNotifications/EXPushTokenModule.m

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXNotifications/EXPushTokenModule.h>
+#import <EXNotifications/EXPushTokenManager.h>
+
+#import <UMCore/UMEventEmitterService.h>
+
+static const NSString *onDevicePushTokenEventName = @"onDevicePushToken";
+
+@interface EXPushTokenModule ()
+
+@property (nonatomic, weak) id<EXPushTokenManager> pushTokenManager;
+
+@property (nonatomic, assign) BOOL isListening;
+@property (nonatomic, assign) BOOL isBeingObserved;
+@property (nonatomic, assign) BOOL isSettlingPromise;
+
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+
+@property (nonatomic, strong) UMPromiseResolveBlock getDevicePushTokenResolver;
+@property (nonatomic, strong) UMPromiseRejectBlock getDevicePushTokenRejecter;
+
+@end
+
+@implementation EXPushTokenModule
+
+UM_EXPORT_MODULE(ExpoPushTokenManager);
+
+# pragma mark - Exported methods
+
+UM_EXPORT_METHOD_AS(getDevicePushTokenAsync,
+                    getDevicePushTokenResolving:(UMPromiseResolveBlock)resolve rejecting:(UMPromiseRejectBlock)reject)
+{
+  if (_getDevicePushTokenRejecter) {
+    reject(@"E_AWAIT_PROMISE", @"Another async call to this method is in progress. Await the first Promise.", nil);
+    return;
+  }
+
+  _getDevicePushTokenResolver = resolve;
+  _getDevicePushTokenRejecter = reject;
+  [self setIsSettlingPromise:YES];
+
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Would user be registered for notification somehow without that? If not, the name of the method seems misleading, since it does much more than it says.

Reply:
i have always had the same concern with this method. if we change it i think that should be done in a followup pr, leaving behavior the same seems better for now.

Reply:
I agree the name doesn't describe all that the function does very clearly, but on the other hand user can't expect to send notifications to a device without calling this method — where would one get the token from?